nativelib/src/main/resources/gc/immix/ImmixGC.c

```diff
@@ -13,8 +14,56 @@
 
 void scalanative_collect();
 
+/*
+ Accepts number of bytes or number with a suffix letter for indicating the units.
+ k or K for kilobytes(1024 bytes), m or M for megabytes and g or G for gigabytes.
+*/
+size_t parseSizeStr(char * str) {
+    int length = strlen(str);
+    size_t size;
+    sscanf(str, "%zu", &size);
+    char possibleSuffix = str[length - 1];
+    switch (possibleSuffix) {
+        case 'k':
+        case 'K':
+            size *= 1024;
+            break;
+        case 'm':
+        case 'M':
+            size *= 1024 * 1024;
+            break;
+        case 'g':
+        case 'G':
+            size *= 1024 * 1024 * 1024;
```

Comment:
This can overflow if size_it is 32-bit. This needs to be checked and reported as an error in case it does.